### PR TITLE
[WIP] Fix read csv sep issue

### PR DESCRIPTION
**Pull request by Iroy30**

### CODE REVIEW

File: cudf/io.py [hunk 1]
```diff
@@ -33,6 +33,8 @@ def read_csv(filepath, lineterminator='\n',
         Delimiter to be used.
     delim_whitespace : bool, default False
         Determines whether to use whitespace as delimiter.
+    sep : char, default None
+        Alternative argument name for delimiter.
```
Comment (randerzander) [inline cudf/io.py]: To match Pandas as closely as possible, this should be:
```suggestion
sep : str, default ‘,’
    Delimiter to use.
```
And:

```suggestion
delimiter : str, default None
     Alternative argument name for sep.
```

File: cudf/io.py [hunk 2]
```diff
@@ -106,6 +108,9 @@ def read_csv(filepath, lineterminator='\n',
     dtype_ptr = ffi.new('char*[]', arr_dtypes)
     csv_reader.dtype = dtype_ptr
 
+    if sep is not None:
+        delimiter = sep
```
Comment (kkraus14) [inline cudf/io.py]: Is this the correct behavior? What if both `delimiter` and `sep` are defined? Should an exception be thrown or does one override the other by default?

**Comment by Iroy30** [comment]
I believe this PR reworks @randerzander PR #388 which fixes the same issue. This will be closed as soon as #388 is approved.

**Comment by harrism** [comment]
Is this PR different from #361 which was just merged?

**Comment by Iroy30** [comment]
yes, it is different from #361. This includes sep argument data which was otherwise ignored. This PR is a subset of #388 and will be closed once that is approved/merged

**Comment by harrism** [comment]
@Iroy30 Can you follow up on this PR and do the following:

- [ ] Respond to @kkraus14 review comments
- [ ] Resolve conflicts
- [ ] Write a description for the PR in the top box.
- [ ] Add a changelog entry in CHANGELOG.md

Code freeze for 0.5 release is in one week.

**Comment by kkraus14** [comment]
Being handled in #604 closing this.

**Pull request closed without merging**